pkg/csi/service/vanilla/controller.go

```diff
@@ -879,11 +896,27 @@ func (c *controller) ControllerUnpublishVolume(ctx context.Context, req *csi.Con
 				},
 			}
 			// Select only the volume type.
-			queryResult, err := c.manager.VolumeManager.QueryAllVolume(ctx, queryFilter, querySelection)
-			if err != nil {
-				msg := fmt.Sprintf("QueryVolume failed for volumeID: %q. %+v", req.VolumeId, err.Error())
-				log.Error(msg)
-				return nil, status.Error(codes.Internal, msg)
+			var queryResult *cnstypes.CnsQueryResult
+			var queryAsyncNotSupported bool
+			if commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.AsyncQueryVolume) {
```

Comment:
Can line 901 to line 919 be refactored into a function and used across this and other files since the same code is repeated in 7-8 different places?

Reply:
I would like to keep it same. The log messages during failures are different and at some places we print function name as a prefix in log messages

Reply:
I dont agree. It doesnt make sense to copy and paste 20 lines of code in 8 different places just for logging. We can log generic statements within the function and have the caller log detailed message to determine the control path. We also already have stack trace for this purpose.

Reply:
Addressed the same in the latest commit: https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/830/commits/fad923dea22e717d86760cb96d5a1a833926b7d9. PTAL